array/array.go

```diff
@@ -206,3 +209,74 @@ func Slice(arr Array, i, j int) Array {
 	err := errors.Newf(codes.Internal, "cannot slice array of type %T", arr)
 	panic(err)
 }
+
+func ToFloatConv(mem memory.Allocator, arr Array) (*Float, error) {
```

Comment:
This method can go in `typeconv.go` itself. Let me know if I have misunderstood something.

Reply:
The thinking was "things that work with arrays directly should likely go in the array package." This sort of puts most of the underlying vectorization work here in one area of the source tree.

Let me know if you think it should really move -- we can try and shift it around if need be.

Reply:
That makes sense.